File: src/main/java/pl/put/poznan/sorting/models/ObjectSortRequest.java
package pl.put.poznan.sorting.models;

import com.google.gson.Gson;
import com.google.gson.JsonSyntaxException;
/**
 * Request sort - json implementation for objects.
 *
 */
public class ObjectSortRequest extends SortRequest {

    /**
     * Assigning arguments to class fields.
     *
     * @param   data
     * @param   algorithms
     * @param   key
     * @param   reverse
     * @param   iterations
     */
    public ObjectSortRequest(Object[] data, String[] algorithms, String key, boolean reverse, int iterations) {
        super(data, algorithms, reverse, iterations);
        this.key = key;
    }

    public final String key;
    /**
     * Reading a request from json.
     *
     * @param   json    type: String
     * @return          sort request
     * @throws JsonSyntaxException
     */
    public static ObjectSortRequest fromJson(String json) throws JsonSyntaxException {
        return new Gson().fromJson(json, ObjectSortRequest.class);
    }

    /**
     * Transorming to json.
     *
     * @return      object transformed to json format.
     */
    public String toJson() {
        return new Gson().toJson(this);
    }

}

File: lib/sort_api/models/object_sort_request.dart
import 'package:json_annotation/json_annotation.dart';

part 'object_sort_request.g.dart';

@JsonSerializable()
class ObjectSortRequest {
  ObjectSortRequest({
    required this.algorithms,
    required this.data,
    required this.key,
    required this.iterations,
    required this.reverse,
  });

  factory ObjectSortRequest.fromJson(Map<String, dynamic> json) =>
      _$ObjectSortRequestFromJson(json);

  final List<String> algorithms;
  final List<dynamic> data;
  final String key;
  final int iterations;
  final bool reverse;

  Map<String, dynamic> toJson() => _$ObjectSortRequestToJson(this);
}
